core/lib/test/unit/hooks.spec.ts

```diff
@@ -0,0 +1,50 @@
+import { createListenerHook, createMutatorHook, createMutatorHookAsync } from '@vue-storefront/core/lib/hooks';
+
+describe('Hooks', () => {
+  describe('createListenerHook', () => {
+    it('executes functions with arguments added by hook', () => {
+      const { hook, executor } = createListenerHook();
+      const mockedFn = jest.fn(arg => arg)
+      const mockedArgs = [1, 2, 3]
+
+      hook(mockedFn)
+      executor(mockedArgs)
+
+      expect(mockedFn).toHaveBeenCalledWith(mockedArgs);
+    })
+  })
+
+  describe('createMutatorHook', () => {
+    it('executes functions added by hook', () => {
+      const { hook, executor } = createMutatorHook();
+      const mockedFn = jest.fn(arg => `${arg} / test`)
+      const mockedRawOutput = 'abc'
+
+      hook(mockedFn)
+      hook(mockedFn)
+      executor(mockedRawOutput)
```

Comment:
You should assign what it returns to some `const` and make an `expect` for that value. So I assume this const should be equal `abc / test`

Reply:
or `abc / test / test`